app/plugin_key_value_store.go

```diff
@@ -23,41 +23,47 @@ func (a *App) SetPluginKey(pluginId string, key string, value []byte) *model.App
 }
 
 func (a *App) SetPluginKeyWithExpiry(pluginId string, key string, value []byte, expireInSeconds int64) *model.AppError {
-
 	if expireInSeconds > 0 {
 		expireInSeconds = model.GetMillis() + (expireInSeconds * 1000)
 	}
 
 	kv := &model.PluginKeyValue{
 		PluginId: pluginId,
-		Key:      getKeyHash(key),
+		Key:      key,
 		Value:    value,
 		ExpireAt: expireInSeconds,
 	}
 
 	result := <-a.Srv.Store.Plugin().SaveOrUpdate(kv)
 
-	if result.Err != nil {
-		mlog.Error(result.Err.Error())
+	// Clean up a previous entry using the hashed key, if it exists.
+	if deleteResult := <-a.Srv.Store.Plugin().Delete(pluginId, getKeyHash(key)); deleteResult.Err != nil {
+		mlog.Error("Failed to clean up previously hashed key", mlog.String("key", key), mlog.Err(deleteResult.Err))
 	}
 
 	return result.Err
 }
 
 func (a *App) GetPluginKey(pluginId string, key string) ([]byte, *model.AppError) {
-	result := <-a.Srv.Store.Plugin().Get(pluginId, getKeyHash(key))
+	result := <-a.Srv.Store.Plugin().Get(pluginId, key)
 
-	if result.Err != nil {
-		if result.Err.StatusCode == http.StatusNotFound {
-			return nil, nil
-		}
+	if result.Err == nil {
+		kv := result.Data.(*model.PluginKeyValue)
+		return kv.Value, nil
+	}
+
```

Comment:
I think 
```go
if result.Err.StatusCode != http.StatusNotFound {
		mlog.Error(result.Err.Error())
		return nil, result.Err
}
```
is missing here.

Reply:
Ah, good catch! I looked at that code, even checked it out, but convinced myself it was ok. I'll patch that on @gupsho's behalf to get this merged today.